meilisearch/index.py

```diff
@@ -397,17 +397,8 @@ def get_documents(
         MeilisearchApiError
             An error containing details about why Meilisearch can't process your request. Meilisearch error codes are described here: https://www.meilisearch.com/docs/reference/errors/error_codes#meilisearch-errors
         """
-        if parameters is None or parameters.get("filter") is None:
-            if parameters is None:
-                parameters = {}
-            elif "fields" in parameters and isinstance(parameters["fields"], list):
-                parameters["fields"] = ",".join(parameters["fields"])
-
-            response = self.http.get(
-                f"{self.config.paths.index}/{self.uid}/{self.config.paths.document}?{parse.urlencode(parameters)}"
-            )
-            return DocumentsResults(response)
```

Comment:
The reason for this is prior to v1.2 the post route for get_documents didn't exist and we wanted to keep compatibility with all v1.x versions. Meilisearch doesn't provide a way to get the version through the api so this is a "best guess". If `filter` is present it has to be v1.2 or greater so the post route is available, if not it could be a version earlier than 1.2, but we don't have any way to know for sure.

You are correct that the `get` route is listed as being depreciated soon. So I understand your issue, and your fix here looks good but it breaks compatibility with <v1.2. One thing we could do is `offset`, `limit`, and `fields` were the only allowed parameters in the older versions so if `parameters` contains any field other than those, i.e. `retrieveVectors` in your case, pass it on to the `post` route. That said it is only a temporary solution and will become a problem again with the `get` route is removed.

 @curquiza thoughts?

Reply:
Ah, I did not think of that, good catch.

Alternatively, we can also scan for `bool` values in the parameters dict and of we find any, force the `POST` route instead of GET?
Or simply try and do some post processing to find `True` and `False` values in the encoded string and replace them, that seems very sketchy to me though, probably prone to misfiring.

Reply:
> You are correct that the get route is listed as being depreciated soon

On our side, we don't plan any v2 in the coming next months, maybe in 2025, but we are not even sure. And if it is, it's more end of 2025. So, the route is not going to be deprecated **_soon_**.

However, I think we can allow our SDK to break sometimes, especially the `POST documents` route is here for a long time. The most important is to be clear
- in the changelog
- in the readme -> we are compatible with v1.X, except if you use theses methods, you need to have v1.2 or later

Reply:
@martinnj lets go with what you have here. We can probably get cleaver and come up with a solution to keep backwards compatibility, but cleaver frequently ends up with bugs so lets keep it simple.

Will you also add a note to the readme about compatibility with v1.2+ as suggested by @curquiza?

Reply:
@sanders41 I was a bit in doubt about where to put it or how to word it, but I've pushed my suggestion.

Let me know if you want something changed and I'll get on it.

Reply:
What you have looks good.